mujoco_ros/src/offscreen_camera.cpp

```diff
@@ -250,6 +250,9 @@ bool OffscreenCamera::renderAndPubIfNecessary(mujoco_ros::OffscreenRenderContext
 
 		depth_pub_.publish(depth_msg);
 	}
+
+	publishCameraInfo(ros::Time(scn_state_.data.time));
```

Comment:
renderAndPubIfNecessary can be called twice when a stream type contains both RGB and SEGMENT. Since they all share the same camera info manager, this means republishing the same data. In case it is necessary to publish camera info without any rendered images it makes more sense making the publish below (in renderAndPublish) unconditional.

Reply:
AFAIK all cameras publish info for each image separately. I could not find any definite answer if that is required anywhere beyond the information in [rep-0104](https://github.com/ros-infrastructure/rep/blob/master/rep-0104.rst#publish-camerainfo-only-when-parameters-change) and the [image_geometry](http://wiki.ros.org/image_geometry) package.